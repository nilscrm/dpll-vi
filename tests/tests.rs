use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
