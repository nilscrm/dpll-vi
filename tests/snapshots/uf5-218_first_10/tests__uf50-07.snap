---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = false
var 2 = false
var 3 = true
var 4 = true
var 5 = false
var 6 = true
var 7 = false
var 8 = false
var 9 = true
var 10 = true
var 11 = true
var 12 = true
var 13 = true
var 14 = false
var 15 = false
var 16 = false
var 18 = false
var 19 = false
var 20 = true
var 21 = true
var 22 = true
var 23 = false
var 24 = false
var 25 = false
var 26 = false
var 27 = true
var 28 = true
var 29 = true
var 30 = true
var 31 = true
var 32 = false
var 33 = false
var 34 = true
var 35 = true
var 36 = false
var 37 = false
var 38 = false
var 39 = false
var 40 = true
var 41 = true
var 42 = true
var 43 = true
var 44 = false
var 45 = false
var 46 = true
var 47 = true
var 48 = false
var 49 = false
var 50 = true

----- stderr -----

Interactions
  Total            8_269_309
    Depth          4_171_666
    Breadth                1
  Annihilate       4_632_919
  Commute             23_975
  Copy               828_469
  Erase              950_896
  Expand             644_476
  Call               755_452
  Branch             433_122

Memory
  Heap               577_600 B
  Allocated      183_270_704 B
  Freed          183_270_704 B

[Performance...]
